### PR TITLE
DRA: e2e_node: improve readability

### PR DESCRIPTION
**Pull request by bart0sh**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

This is a follow-up PR to address [this review comment](https://github.com/kubernetes/kubernetes/pull/124617#discussion_r1643938580)

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

**Review by pohly — APPROVED** [review]
/lgtm

**Comment by bart0sh** [comment]
/assign @SergeyKanzhelev @mrunalp 
for approval

File: test/e2e_node/dra_test.go [hunk 1]
```diff
@@ -178,7 +178,7 @@ var _ = framework.SIGDescribe("node")("DRA", feature.DynamicResourceAllocation,
 			ginkgo.By("wait for NodePrepareResources call to fail")
 			gomega.Eventually(kubeletPlugin.GetGRPCCalls).WithTimeout(retryTestTimeout).Should(testdriver.NodePrepareResourcesFailed)
 
-			unset()
+			unsetNodePrepareResourcesFailureMode()
```
Comment (SergeyKanzhelev) [inline test/e2e_node/dra_test.go:181]: does this needs to be in defer as well? In case test will fail in the middle?

Comment (bart0sh) [reply test/e2e_node/dra_test.go:181]: As kubeletPlugin is created for every test case it's safe enough not to defer in my opinion.

**Pull request merged**